src/App/pages/ReposSearchPage/components/Search/components/Button/Button.tsx

```diff
@@ -0,0 +1,20 @@
+import './Button.scss';
+import React from "react";
+
+import SearchIcon from "./components/SearchIcon";
+
+type ButtonProps = {
+    onClick?: (e: React.MouseEvent) => void;
+    disabled: boolean;
+}
+
+const Button: React.FC<ButtonProps> = ({onClick, disabled}) => {
+
+    return (
+        <button className="button" onClick={disabled ? () => 0 : onClick}>
```

Comment:
можно просто передать disabled атрибут, при значении `true`, `onClick` не будет срабатывать
```suggestion
        <button className="button" disabled={disabled} onClick={onClick}>
```